File: AMF5/src/test/java/javaPlatform/JSONSchemaParsingTest.java
package javaPlatform;

import amf.aml.client.platform.AMLBaseUnitClient;
import amf.core.client.platform.AMFParseResult;
import amf.core.client.platform.model.document.BaseUnit;
import amf.core.client.platform.model.domain.Shape;
import amf.shapes.client.platform.ShapesConfiguration;
import amf.shapes.client.platform.config.JsonSchemaConfiguration;
import amf.shapes.client.platform.model.document.JsonSchemaDocument;
import org.junit.Test;

import java.util.List;
import java.util.concurrent.ExecutionException;

import static org.junit.Assert.*;

public class JSONSchemaParsingTest {

    @Test
    public void parseJSONSchema() throws ExecutionException, InterruptedException {
        ShapesConfiguration shapesConfiguration = JsonSchemaConfiguration.JsonSchema();
        final AMLBaseUnitClient client = shapesConfiguration.baseUnitClient();

        // A BaseUnit is the return type of any parsing
        // The actual object can be many different things, depending on the content of the source file
        // https://github.com/aml-org/amf/blob/develop/documentation/model.md#baseunit
        final AMFParseResult parseResult = client.parse("file://src/test/resources/examples/schema.json").get();
        final BaseUnit model = parseResult.baseUnit();

        assertTrue(parseResult.conforms());
        assertNotNull(model);

        // In this case the BaseUnit is a JsonSchemaDocument
        assertSame(model.getClass(), JsonSchemaDocument.class);

        final JsonSchemaDocument document = (JsonSchemaDocument) model;

        // A JsonSchemaDocument encodes a Shape, which is the root schema of the JSON Schema
        final Shape rootShape = (Shape) document.encodes();
        assertNotNull(rootShape);

        // A JsonSchemaDocument could declares a list of Shapes, which are the schemas inside the `definitions` or `$defs` key
        @SuppressWarnings("unchecked")
        final List<Shape> declarations = (List<Shape>) (Object) document.declares();

        assertFalse(declarations.isEmpty());
    }
}

File: AMF5/src/test/scala/scalaPlatform/JSONSchemaParsingTest.scala
package scalaPlatform

import amf.core.client.scala.model.domain.Shape
import amf.shapes.client.scala.config.JsonSchemaConfiguration
import amf.shapes.client.scala.model.document.JsonSchemaDocument
import org.junit.Assert.{assertEquals, assertNotNull, assertTrue}
import org.junit.runner.RunWith
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
import org.scalatest.matchers.should
import org.scalatestplus.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class JSONSchemaParsingTest extends AsyncFlatSpec with should.Matchers {

  it should "parse JSON Schema file" in {
    val client = JsonSchemaConfiguration.JsonSchema().baseUnitClient()

    // A BaseUnit is the return type of any parsing
    // The actual object can be many different things, depending on the content of the source file
    // https://github.com/aml-org/amf/blob/develop/documentation/model.md#baseunit
    client.parse("file://src/test/resources/examples/schema.json") map { result =>

      assertNotNull(result.baseUnit)

      // In this case the BaseUnit is a JsonSchemaDocument
      result.baseUnit mustBe a[JsonSchemaDocument]
      val document = result.baseUnit.asInstanceOf[JsonSchemaDocument]

      // A JsonSchemaDocument encodes a Shape, which is the root schema of the JSON Schema
      val rootShape: Shape = document.encodes
      assertNotNull(rootShape)

      // A JsonSchemaDocument could declares a list of Shapes, which are the schemas inside the `definitions` or `$defs` key
      val declarations: Seq[Shape] = document.declares.map(de => de.asInstanceOf[Shape])

      assertTrue(declarations.nonEmpty)
      assertEquals(declarations.size, 1)

      result.conforms shouldBe true
    }
  }
}
